fix(go-ci): git diff fails with symmetric difference error on shallow clone

## Problem

In the `lint` job of `tpl-go-ci.yml`, the "Check for changed Go files" step runs:

```bash
git diff --name-only --diff-filter=ACMR "$PR_BASE_SHA"...HEAD -- "*.go"
```

On a shallow clone (default for `actions/checkout@v4`), the base commit is not present in local history. This causes:

```
fatal: Invalid symmetric difference expression fa973ddd...HEAD
Error: Process completed with exit code 128.
```

## Fix

Fetch the base SHA before the diff step:

```yaml
- name: Fetch base SHA
  env:
    PR_BASE_SHA: ${{ github.event.pull_request.base.sha }}
  run: git fetch --depth=1 origin "$PR_BASE_SHA"
```

This ensures the base commit is available for the three-dot diff.

--- a/.github/workflows/tpl-go-ci.yml
+++ b/.github/workflows/tpl-go-ci.yml
@@ -95,6 +95,11 @@ jobs:
           go-version-file: ${{ inputs.go-version-file }}
           working-directory: ${{ inputs.working-directory }}
 
+      - name: Fetch base SHA
+        env:
+          PR_BASE_SHA: ${{ github.event.pull_request.base.sha }}
+        run: git fetch --depth=1 origin "$PR_BASE_SHA"
+
       - name: Check for changed Go files
         id: go-files
         env:

